apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -53,3 +60,24 @@ func validateRAID(r *RAIDConfig) error {
 
 	return nil
 }
+
+func validateDNSName(hostaddress string) error {
+
+	if hostaddress == "" {
+		return nil
+	}
+
+	hurl, _ := url.Parse(hostaddress)
+	host, _, err := net.SplitHostPort(hurl.Host)
+
+	if err != nil {
```

Comment:
Maybe error is not nil actually means that host is invalid?

Reply:
Yes, I should probably check if the port is empty rather than the error.

Reply:
İf error is not nil, we are setting `host = hurl.host`, why we are doing that instead returning an error?.